### PR TITLE
reordering of global-attribute in text and full cddl

### LINKED ISSUE #29: can't use CDDL to validate seemingly valid CBOR
If I try to validate the following minimalist (and seemingly valid) `file-entry` using the CDDL provided in the draft:
```
# diag
{ 24: "fw.bin" }

# pretty
a1                 # map(1)
   18 18           # unsigned(24)
   66              # text(6)
      66772e62696e # "fw.bin"
```
I get the following error from the [cddl](https://rubygems.org/gems/cddl) tool:
```
$ cddl file-entry.cddl validate min.cbor
CDDL validation failure (nil for {24=>"fw.bin"}):
["fw.bin", [:prim, 3], nil]
["fw.bin", [:prim, 3], null]
```
This happens because the inlined `any-attribute` consumes all of the map, leaving nothing for `fs-name => text` (this is what the rather cryptic message from the cddl tool is about.)

One could fix this specific bit of CDDL by relocating `global-attributes` to the bottom of `filesystem-item`, but given the pervasiveness of the `global-attributes` group embedding, it's almost certain that similar situations will arise in other places too depending on the relative positioning of the various pieces.

My suggestion would be to make `global-attributes` a map instead, in order remove ambiguities by enforcing tighter namespace segregation.

Note that SWID uses explicit namespacing to cleanly handle these kinds of conflicts.  For example, the following is valid SWID because the `foofoo` extension is explicitly provided under a `tns` namespace:
```
<SoftwareIdentity
  xmlns="http://standards.iso.org/iso/19770/-2/2015/schema.xsd"
  xmlns:tns="http://localhost/foo.xsd"
  name="Roadrunner boot loader"
  tagId="example.acme.roadrunner-sw-bl-v1-0-0"
  version="1.0.0"
  tns:foofoo="bing"
>
[...]
</SoftwareIdentity>
```
Whereas this non-namespaced `foofoo` (closely matching the `any-attribute` philosophy) would fail validation:
```
<SoftwareIdentity
  xmlns="http://standards.iso.org/iso/19770/-2/2015/schema.xsd"
  name="Roadrunner boot loader"
  tagId="example.acme.roadrunner-sw-bl-v1-0-0"
  version="1.0.0"
  foofoo="bong"
>
[...]
</SoftwareIdentity>
```

So more generally it seems to me that the way `global-attributes` are currently handled is way too lax and doesn't reflect the original SWID approach.

### PR DESCRIPTION
**Pull request by henkbirkholz**
fixes #29

### CODE REVIEW

File: concise-swid-tag.cddl [hunk 1]
```diff
@@ -12,16 +12,14 @@ concise-swid-tag = {
   entity => entity-entry / [ 2* entity-entry ],
   ? link => link-entry / [ 2* link-entry ],
   ? payload-or-evidence,
-  global-attributes,
   * $$coswid-extension,
+  global-attributes,
 }
 
 payload-or-evidence //= ( payload => payload-entry )
-payload-or-evidence //= ( payload => [ 2* payload-entry ] )
```
Comment (cabo) [inline concise-swid-tag.cddl:20]: Is this an unrelated change?

File: concise-swid-tag.cddl [hunk 2]
```diff
@@ -165,16 +163,16 @@ filesystem-item = (
...
   resource-collection,
-  global-attributes,
   * $$payload-extension,
+  global-attributes,
 }
 
 evidence-entry = {
   resource-collection,
-  ? date => time,
+  ? date => #6.1(int),
```
Comment (cabo) [inline concise-swid-tag.cddl]: I think it would be better style to add a

```
integer-time = #6.1(int)
```

and reference that here.

File: concise-swid-tag.cddl [hunk 3]
```diff
@@ -12,16 +12,14 @@ concise-swid-tag = {
...
+  global-attributes,
 }
 
 payload-or-evidence //= ( payload => payload-entry )
-payload-or-evidence //= ( payload => [ 2* payload-entry ] )
 payload-or-evidence //= ( evidence => evidence-entry )
-payload-or-evidence //= ( evidence => [ 2* evidence-entry ] )
 
-any-uri = text
+any-uri = uri
```
Comment (cabo) [inline concise-swid-tag.cddl:22]: Unrelated change?

File: concise-swid-tag.cddl (hunk 1 above)
Comment (henkbirkholz) [reply concise-swid-tag.cddl:20]: yes, that is a change that leaked over from the "isomorphic issue"

File: concise-swid-tag.cddl (hunk 3 above)
Comment (henkbirkholz) [reply concise-swid-tag.cddl:22]: ack

File: concise-swid-tag.cddl (hunk 2 above)
Comment (henkbirkholz) [inline concise-swid-tag.cddl]:
```suggestion
  ? date => integer-time,
```
